js/main.js

```diff
@@ -404,39 +404,56 @@ function initiate() {
             );
         }
 
+        function addAllTypicalWarningsInFragment(fragment, fragmentIndex, badness, errorCode, type) {
```

Comment:
`badness` is a regexp,isn't it? Can you rename it to `badnessRe` and/or add `instanceof` assertion?